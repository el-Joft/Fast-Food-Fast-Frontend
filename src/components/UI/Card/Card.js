/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getAllMenus } from '../../../actions/getMenusAction';
import { addCart } from '../../../actions/orderAction';


class Card extends Component {
  componentDidMount() {
    this.props.getAllMenus();
  }

  addToCart(menuData) {
    const item = Object.assign({}, menuData, { quantity: 1 });
    this.props.addCart(item);
  }

  renderCard = () => {
    const { menu } = this.props;
    if (menu.menu !== null) {
      return menu.menu.map(item => (
        <div key={item.id} className='menu-section'>
          <div className="menu-food">
            <h3>{item.name}</h3>
            <p>{item.description}</p>
          </div>
          <div className="menu-price">
            <p>
              <strike>N</strike>
              {item.price}
            </p>
            <div to="/checkout">
              {' '}
              <span
                role="searchbox"
                onKeyPress={() => this.addToCart(item)}
                onClick={menuData => this.addToCart(item)}
                tabIndex="-1"
                className="menu-order"
              >
                Make Order
              </span>
            </div>
          </div>

        </div>
      ));
    }
  }

  render() {
    return (
      <div className="menu-body">
        {this.renderCard()}
      </div>
    );
  }
}

Card.propTypes = {
  getAllMenus: PropTypes.func.isRequired,
  addCart: PropTypes.func.isRequired,
  menu: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired,
};

// we get values here using the this.props.errors/auth

const mapStateToProps = state => ({
  menu: state.menu,
  errors: state.errors,
  order: state.order,
});

export default connect(mapStateToProps, { getAllMenus, addCart })(Card);
