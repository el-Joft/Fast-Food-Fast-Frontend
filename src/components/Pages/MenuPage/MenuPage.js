/* eslint-disable import/no-named-as-default */
import React from 'react';
import Card from '../../UI/Card/Card';

export const MenuPage = () => (
  <div>
    <section className="container">
      <div className="menu-banner">
        <div>
          <div className="banner-text">
            <h2>Available Menu</h2>
            <p>Learn about our dishes</p>
          </div>
        </div>
      </div>
    </section>
    <Card />
  </div>
);

export default MenuPage;
