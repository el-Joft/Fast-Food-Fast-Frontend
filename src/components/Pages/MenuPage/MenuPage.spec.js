/* eslint-disable react/react-in-jsx-scope */
import { shallow } from 'enzyme';
import { MenuPage } from './MenuPage';
import { Card } from '../../UI/Card/Card';

describe('rendering', () => {
  let wrapper;
  it('should render the component', () => {
    wrapper = shallow(<MenuPage />);
    expect(wrapper.length).toBe(1);
  });

  it('should render the component', () => {
    wrapper = shallow(<MenuPage />);
    expect(wrapper.find(Card)).toEqual({});
  });

  it('should render the component', () => {
    wrapper = shallow(<MenuPage />);
    expect(wrapper).toMatchSnapshot();
  });
});
