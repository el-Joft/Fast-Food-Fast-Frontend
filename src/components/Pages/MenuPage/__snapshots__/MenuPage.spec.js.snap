// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render the component 1`] = `
<div>
  <section
    className="container"
  >
    <div
      className="menu-banner"
    >
      <div>
        <div
          className="banner-text"
        >
          <h2>
            Available Menu
          </h2>
          <p>
            Learn about our dishes
          </p>
        </div>
      </div>
    </div>
  </section>
  <Connect(Card) />
</div>
`;
